main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentController.cs

```diff
@@ -384,12 +384,26 @@ public abstract class DocumentController : IDisposable, ICommandDelegatorRouter
 			}
 		}
 
+		CancellationTokenSource disposedTokenSource = new CancellationTokenSource ();
+
+		/// <summary>
+		/// Returns a dispose token that's canceled when the document controller disposes.
+		/// </summary>
+		protected CancellationToken DisposedToken { get; private set; }
```

Comment:
```suggestion
		protected CancellationToken DisposedToken { get; }
```

We never change it, so let's not set it anywhere else except the constructor.